src/services/utils.js

```diff
@@ -1,3 +1,14 @@
 export const toGqlString = str => str.replace('.', '__');
 export const fromGqlString = str => str.replace('__', '.');
 export const hocToRenderProps = hoc => hoc(({ render, ...props }) => render(props));
+export const memoize = asyncOp => {
+  const hash = JSON.stringify;
+  const cache = {};
+  return async (...input) => {
+    const hasKey = hash(input);
+    if (!cache[hasKey]) {
+      cache[hasKey] = await asyncOp(...input);
+    }
+    return cache[hasKey];
+  };
+};
```

Comment:
@ciaranschutte just wanna note on this potentially useful function :P didn't really wanna bring in whole libraries for this